crates/metal-ast/src/expr.rs

```diff
@@ -1,7 +1,16 @@
-use crate::Ident;
+use crate::{Ident, Path, Ty};
 
 /// An expression, also called a value.
 pub enum Expr<'src> {
     /// An identifier in place of an expression, like `arg` in `print(arg)`.
     Ident(Ident<'src>),
+    // TODO: represent this some other way
+    /// A number value, like `1` or `2.6`.
+    Number { ty: Ty<'src>, value: u64 },
+    // A function call
+    FnCall {
+        fn_name: Ident<'src>,
+        arguments: Vec<Expr<'src>>,
+        module_name: Path<'src>,
```

Comment:
It's a module name. i.e. you do an import like:

`import a.b.{c};`

the module name would be `a.b` which is used in the LLVM IR function name (if it's not `main.mt`)

Reply:
Oh, right. In that case, it'll best to merge `fn_name` and `module_name` into one, but that'll be done in another PR i didn't push yet

Reply:
IMO, keep it this way, it'd probably be easier to debug in the long run

Reply:
Since AST is the generalization of Metal code, it's sensible to represent both `print("hi")` and `std.io.print("hi")` with one, unified "path" structure, often a vec of idents but more generally an expression on it's own (to allow things like `(self.callbacks.get(...))(arg1, arg2)`. It's not uncommon do it this way, such as [Python](https://docs.python.org/3/library/ast.html#ast.Call), [Rust](https://doc.rust-lang.org/beta/nightly-rustc/rustc_ast/ast/enum.ExprKind.html#variant.Call), [Oxc (a JS/TS toolkit)](https://docs.rs/oxc/latest/oxc/ast/ast/struct.CallExpression.html), [clang (C/C++)](https://clang.llvm.org/doxygen/classclang_1_1CallExpr.html) and i suspect many more. It also aligns better with how the parser works.